source/extensions/tracers/opentelemetry/opentelemetry_tracer_impl.cc

```diff
@@ -26,16 +29,26 @@ Driver::Driver(const envoy::config::trace::v3::OpenTelemetryConfig& opentelemetr
       tracing_stats_{OPENTELEMETRY_TRACER_STATS(
           POOL_COUNTER_PREFIX(context.serverFactoryContext().scope(), "tracing.opentelemetry"))} {
   auto& factory_context = context.serverFactoryContext();
+
+  if (opentelemetry_config.has_grpc_service() && opentelemetry_config.has_http_service()) {
```

Comment:
This seems correct, since the factory method invokes this directly, so it should throw during config ingest.

Reply:
You mean the validation makes sense but here is not the correct place? Could you point me where "config ingest" happens/where should I move this check?

Reply:
Sorry,  comment was not actionable, it was reasoning to myself and other reviewers why what you have there is safe :)